### ISSUE TITLE
Allow completely disabling/hiding default templates

### ISSUE THREAD

**Issue by LuckynaSan** [swissspidy/web-stories-wp#10124]
## Feature Description
A user has requested a way to customize the default view for the Templates dashboard. Currently, the Templates dashboard loads all templates, but this feature request would allow a customized / filtered view of the templates on load.

## Additional Context

related support topic: https://wordpress.org/support/topic/remove-default-story-and-page-templates/

-----

## Acceptance Criteria

* Pass`'canViewDefaultTemplates' => true` to app in `\Google\Web_Stories\Admin\Editor::get_editor_settings()` and `\Google\Web_Stories\Admin\Dashboard::get_dashboard_settings()`
* In dashboard, hide "Explore Templates" from menu if user does not have capability
* In dashboard, prevent direct access to templates if user does not have capability (feasible?)
* In editor, hide "Default Templates" from the page templates tab, hide the selection dropdown, and always show the custom templates.

This will allow devs to disable access to default templates via the `web_stories_editor_settings` and `web_stories_dashboard_settings` filters.

**Alternative:**

Use something like `current_user_can( 'web_stories_view_default_templates' )` instead, which allows devs to filter this via `map_meta_cap`.

It's a bit more verbose which helps a lot with comprehension.

We could even do the same for the `showMedia3p` config.

**Comment by swissspidy** [comment]
So the question is really to list saved page templates in the dashboard?

The dashboard only shows the default templates, not saved page templates.

Default templates are essentially full stories consisting of multiple pages, plus a description, etc.

On the other hand, custom saved templates are only really a single page, without any other information. Showing these in the dashboard would therefore require UX.

The support topic also mentions the editor.

I'd suggest mentioning to the user that once they select "Saved Templates" in the editor, that selection will persist the next time they load the editor:

![Screenshot 2022-01-07 at 17 16 35](https://user-images.githubusercontent.com/841956/148573057-a46ea1de-e514-4233-adae-6f8b0d55138e.png)

I think knowing that should solve their request.

**Comment by LuckynaSan** [comment]
This user wants a way to either remove or hide the default templates from the templates dashboard, as well as any default page templates from the page templates panel.

**Comment by lucymtc** [comment]
Thanks for creating this issue. 
The main purpose of this request is so editors stick as much as possible to a certain style guides, therefore we don't want to have any of the default story templates in the dashboard as well as removing the default page templates from the editor. 

> I'd suggest mentioning to the user that once they select "Saved Templates" in the editor, that selection will persist the next time they load the editor

What we are looking for is to not have the Default templates option at all. 

I was looking for an option in the dashboard and editor settings that could be filtered to true or false, or maybe filter the list of the actual templates to exclude/include the desired ones.
I hope this clarifies it a bit more.

**Comment by swissspidy** [comment]
Thanks a lot, that definitely does clarify it!

It sounds like we might be able to achieve this using a custom (meta) capability and allow developers to grant access for the default templates via code. That'd allow disabling/hiding them completely. Does that work for your use case or would it need to be a UI setting?

**Comment by lucymtc** [comment]
@swissspidy Yes, that sounds good. Thanks!

**Comment by swissspidy** [comment]
@lucymtc ICYMI, this is how the default templates can now be disabled on the dashboard and in the editor:

```php
function gh_10124_change_get_settings( array $settings ): array {
	$settings['canViewDefaultTemplates'] = false;

	return $settings;
}
add_filter( 'web_stories_editor_settings', 'gh_10124_change_get_settings' );
add_filter( 'web_stories_dashboard_settings', 'gh_10124_change_get_settings' );
```

Available since 1.17.0
